### DISCUSSION TITLE
Name field for ls on an absolute path

### DISCUSSION THREAD

**Discussion by tesujimath** [nushell/nushell#13283] [Ideas]
`ls` on an absolute path behaves quite differently from traditional shells, and in some ways is less ergonomic.  Specifically I think inclusion of the full path in the name field is unhelpful.

Simple example:
```
it23677> ls /nix/store/nnvsjd3f3dh9wdl4s9mwg5cfri8kds5j-bash-interactive-5.2p26
╭────┬────────────────────────────────────────────────────────────────────────────┬───────┬──────────┬──────────────╮
│  # │                                    name                                    │ type  │   size   │   modified   │
├────┼────────────────────────────────────────────────────────────────────────────┼───────┼──────────┼──────────────┤
│  0 │ /nix/store/nnvsjd3f3dh9wdl4s9mwg5cfri8kds5j-bash-interactive-5.2p26/lib    │ dir   │  4.0 KiB │ 54 years ago │
│  1 │ /nix/store/nnvsjd3f3dh9wdl4s9mwg5cfri8kds5j-bash-interactive-5.2p26/share  │ dir   │  4.0 KiB │ 54 years ago │
│  2 │ /nix/store/nnvsjd3f3dh9wdl4s9mwg5cfri8kds5j-bash-interactive-5.2p26/bin    │ dir   │  4.0 KiB │ 54 years ago │
╰────┴────────────────────────────────────────────────────────────────────────────┴───────┴──────────┴──────────────╯
it23677> bash -c "ls /nix/store/nnvsjd3f3dh9wdl4s9mwg5cfri8kds5j-bash-interactive-5.2p26"
bin  lib  share
```
I've picked a tiny directory, but look how much screen real-estate is being given to showing the full path, over and over again.

So I understand that path globbing is a thing in Nu, and so we need to be able to show files from different directories, e.g.
```
it23677> ls /nix/store/nnvsjd3f3dh9wdl4s9mwg5cfri8kds5j-bash-interactive-5.2p26/**/* | take 10
╭────┬──────────────────────────────────────────────────────────────────────────────────────────────┬─────────┬─────╮
│  # │                                             name                                             │  type   │ ... │
├────┼──────────────────────────────────────────────────────────────────────────────────────────────┼─────────┼─────┤
│  0 │ /nix/store/nnvsjd3f3dh9wdl4s9mwg5cfri8kds5j-bash-interactive-5.2p26/bin                      │ dir     │ ... │
│  1 │ /nix/store/nnvsjd3f3dh9wdl4s9mwg5cfri8kds5j-bash-interactive-5.2p26/bin/bash                 │ file    │ ... │
│  2 │ /nix/store/nnvsjd3f3dh9wdl4s9mwg5cfri8kds5j-bash-interactive-5.2p26/bin/bashbug              │ file    │ ... │
│  3 │ /nix/store/nnvsjd3f3dh9wdl4s9mwg5cfri8kds5j-bash-interactive-5.2p26/bin/sh                   │ symlink │ ... │
│  4 │ /nix/store/nnvsjd3f3dh9wdl4s9mwg5cfri8kds5j-bash-interactive-5.2p26/lib                      │ dir     │ ... │
│  5 │ /nix/store/nnvsjd3f3dh9wdl4s9mwg5cfri8kds5j-bash-interactive-5.2p26/lib/bash                 │ dir     │ ... │
│  6 │ /nix/store/nnvsjd3f3dh9wdl4s9mwg5cfri8kds5j-bash-interactive-5.2p26/lib/bash/Makefile.sample │ file    │ ... │
│  7 │ /nix/store/nnvsjd3f3dh9wdl4s9mwg5cfri8kds5j-bash-interactive-5.2p26/lib/bash/accept          │ file    │ ... │
│  8 │ /nix/store/nnvsjd3f3dh9wdl4s9mwg5cfri8kds5j-bash-interactive-5.2p26/lib/bash/basename        │ file    │ ... │
│  9 │ /nix/store/nnvsjd3f3dh9wdl4s9mwg5cfri8kds5j-bash-interactive-5.2p26/lib/bash/csv             │ file    │ ... │
╰────┴──────────────────────────────────────────────────────────────────────────────────────────────┴─────────┴─────╯
```
but wondering whether there is another approach here worth exploring?

```
╭────┬──────────────────────────┬─────────┬─────╮
│  # │ name                     │  type   │ ... │
├────┼──────────────────────────┼─────────┼─────┤
│  0 │ bin                      │ dir     │ ... │
│  1 │ bin/bash                 │ file    │ ... │
│  2 │ bin/bashbug              │ file    │ ... │
│  3 │ bin/sh                   │ symlink │ ... │
│  4 │ lib                      │ dir     │ ... │
│  5 │ lib/bash                 │ dir     │ ... │
│  6 │ lib/bash/Makefile.sample │ file    │ ... │
│  7 │ lib/bash/accept          │ file    │ ... │
│  8 │ lib/bash/basename        │ file    │ ... │
│  9 │ lib/bash/csv             │ file    │ ... │
╰────┴──────────────────────────┴─────────┴─────╯
```

With some careful thought here I wonder if there's a way to make the common cases more usable.

This occurred to me when I was writing this fragment:
```
aya> ls -l $"/proc/($nu.pid)" | where {|it| ($it.name | path basename) == cwd} | select name type target
╭───┬──────────────────┬─────────┬───────────╮
│ # │       name       │  type   │  target   │
├───┼──────────────────┼─────────┼───────────┤
│ 0 │ /proc/186928/cwd │ symlink │ /home/sjg │
╰───┴──────────────────┴─────────┴───────────╯
```
I didn't enjoy having to use `path basename` in there to pick apart the name.  Perhaps there should be separate name and path fields?

Why do I even open such a discussion, rather than just accepting that this is the way things are in Nu?  Because having been using bash for more than 30 years, I feel I may be using Nu for the next 30, it's that good.  And therefore worth a bit of tweaking to make sure it's as good as it could possibly be.  Hope that's OK 😄

**Comment by ghost** [comment]
My 2 cents:

as a nix user im also annoyed by the long prefixes (especially when debugging `nix-build` for python packages (`/nix/store/kg6jlgx4w4qm1amjih25rcldf63pdhvy-python3.11-requests-2.32.3/lib/python3.11/site-packages/requests`)).  
but i think that's a nix-user problem and other distros usually don't use absolute paths (especially this long) and don't mind adding `-s` every once in a while.

a potential solution: add a config variable to make `--short-names` a default argument (which can be overwritten by `--full-paths`).  
another side-note here: `--relative-paths` might be helpful as well

> So I understand that path globbing is a thing in Nu, and so we need to be able to show files from different directories

suggestion: globbing causes the current behavior (relative if not starting with `/` and full otherwise) and otherwise file name only?

> Perhaps there should be separate name and path fields?

if that's the default it would be annoying to always join it (`(ls */*.mp4).name | each` for example is a common use-case for me).  
as a argument (`--seperate-path-and-name`) it could be helpful sometimes
